### PR TITLE
Sources

### PR DESCRIPTION
**Pull request by mphprogrammer**
Added sources resource and specs.

### CODE REVIEW

File: app/controllers/raw_file_sources_controller.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	end
+
+	def edit
+		respond_to do |format|
+			format.html
+		end	
+	end
+
+	def update
+    respond_to do |format|
```
Comment (plusjeff) [inline app/controllers/raw_file_sources_controller.rb]: Nitpick: Formatting seems off here. Is the linter running? (I think it'll auto correct this)

File: app/views/raw_file_sources/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<h1>Raw File Sources</h1>
+<p>
+	This are the allowed raw file sources for the GIBCT. These sources reflect
```
Comment (plusjeff) [inline app/views/raw_file_sources/new.html.erb]: "These" are...

File: app/views/raw_file_sources/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<div class="row">
+	<div class="col-xs-8">
+		This are the allowed raw file sources for the GIBCT. These sources reflect
```
Comment (plusjeff) [inline app/views/raw_file_sources/show.html.erb]: "These" are ...

File: spec/controllers/raw_file_sources_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	describe "GET new" do
+		login_user
+		render_views
+
+		before(:each) do
+			create :raw_file_source
+			get :new
+		end
+
+		it "renders the new page" do
```
Comment (plusjeff) [inline spec/controllers/raw_file_sources_controller_spec.rb]: Does this test just check to see if the response type is HTML? Does it need to be checking other things to verify that data is being pulled correctly?

**Comment by jeffmaher** [comment]
Could you update the README to describe setup? A possible template is:

https://github.com/department-of-veterans-affairs/roadrunner-rails/blob/master/README.template.md

**Comment by jeffmaher** [comment]
From a user standpoint, what does someone do with this screen? Could you describe what will happen in the end system when someone adds a new source?

image

**Comment by jeffmaher** [comment]
Bouncing back for feedback and responses. Toss back my way when done.

**Comment by mphprogrammer** [comment]
@plusjeff That screen is for me to set the build order, for each data source type. The order in which the csv's are processed and built in the database will matter. It (the build order) may change as things change down the road.

**Comment by mphprogrammer** [comment]
@plusjeff The "GET new" spec just checks whether or not a page is rendered and that it contains a form. There really is no other functionality. The new controller creates an empty raw_file_source instance for the form_for helper in the view.

**Comment by mphprogrammer** [comment]
@plusjeff I will update the read me in a separate pull (coming in a few minutes). The seeds.rb has login information (it needs to create a devise logins for development uses on staging and locally, and eventually production) and cannot be put onto github. I will write a simple rake task to generate a shell seeds file without the login information an put instructions to run the task and add their own logins.

**Comment by mphprogrammer** [comment]
@plusjeff I ran brakeman locally, and it complained about ActionPack 4.2.5.1, suggesting that I update to 4.2.5.2 - I will do so by using rails 4.2.5.2 (currently 4.2.5.1).

**Pull request merged**